packaging/package.sh

```diff
@@ -0,0 +1,267 @@
+    fi
+
+    # Get from target
+    if ! ${binPreparedFlag}; then
+        if $(cp -rf ../target/bin/${BIN_NAME} ./ &>/dev/null); then
+            binPreparedFlag=true
+        fi
+    fi
+
+    # Get from Apache Dist
+    if ! ${binPreparedFlag}; then
+        if $(curl -O https://archive.apache.org/dist/ignite/${IGNITE_VERSION}/${BIN_NAME} &>/dev/null); then
+            binPreparedFlag=true
+        else
+           rm -rf ${BIN_NAME} 
+        fi
+    fi
+
+    # Fail if none of the above acquiring method succeeded
+    if ! ${binPreparedFlag}; then
+        echo "[ERROR] Can't find | get Apache Ignite's binary archive"
+        exit 1
+    fi
+}
+
+
+# Build RPM package
+buildRPM () {
+    RPM_WORK_DIR="$(mktemp -d)"
+    rm -rfv *.rpm
+
+    # Prepare build layout
+    mkdir -pv ${RPM_WORK_DIR}/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+    cp -rfv ${BIN_NAME} rpm/{name.service,service.sh} ${RPM_WORK_DIR}/SOURCES
+    cp -rfv rpm/apache-ignite.spec ${RPM_WORK_DIR}/SPECS
+
+    # Assemble RPM packages
+    rpmbuild -bb -v --define "_topdir ${RPM_WORK_DIR}" ${RPM_WORK_DIR}/SPECS/apache-ignite.spec
+
+    # Gather RPMS
+    find ${RPM_WORK_DIR} -name "*.rpm" -exec mv -fv {} ${PACKAGING_DIR} \;
+}
+
+
+# Build DEB package
+buildDEB () {
+    DEB_WORK_DIR="$(mktemp -d)"
+    rm -rfv *.deb
+
+    # Check that RPMs for building exists
+    packageVersion="$(cat deb/changelog | head -1 | sed -r 's|.*\((.*)\).*|\1|')"
+    if [ ! -f "apache-ignite-${packageVersion}.noarch.rpm" ]; then
+        echo "[ERROR] RPM for converting to DEB not found"
+        exit 1
+    fi
+
+    # Unpack RPMs and prepare DEBs build layout
+    cd ${DEB_WORK_DIR}
+    cp -rfv ${PACKAGING_DIR}/apache-ignite-${packageVersion}.noarch.rpm ${DEB_WORK_DIR}
+    ${SUDO} alien --scripts --verbose --keep-version --single apache-ignite-${packageVersion}.noarch.rpm
+
+    # Copy custom DEBs control files and make some modifications on the fly
+    buildDirVersion="$(echo ${packageVersion} | cut -f1 -d-)"
+    cp -rfv ${PACKAGING_DIR}/deb/{changelog,control,copyright,rules} ${DEB_WORK_DIR}/apache-ignite-${buildDirVersion}/debian
+    sed -i -r -e 's|/usr/bin/mkdir|/bin/mkdir|' -e 's|/usr/bin/chown|/bin/chown|' ${DEB_WORK_DIR}/apache-ignite-${buildDirVersion}/etc/systemd/system/apache-ignite@.service
+
+    # Assemble DEB packages
+    cd ${DEB_WORK_DIR}/apache-ignite-${buildDirVersion}
+    ${SUDO} debian/rules binary
+
+    # Gather DEBs
+    find ${DEB_WORK_DIR} -name "*.deb" -exec mv -fv {} ${PACKAGING_DIR} \;
+}
+
+
+# Trap function
+processTrap () {
+    # Removing temporary files
+    echo "Removing temporary work directories: ${DEB_WORK_DIR} ${RPM_WORK_DIR}"
+    rm -rf ${DEB_WORK_DIR} ${RPM_WORK_DIR}
+
+    # Finish
+    echo
+    TIME="$(($(date +%s) - START_TIME))"
+    echo "=== Run time: $(printf '%dh:%02dm:%02ds\n' $((TIME/3600)) $((TIME%3600/60)) $((TIME%60))) ==="
+    echo
+}
+
+
+
+###########
+#  START  #
+###########
+START_TIME=$(date +%s)
+clear
```

Comment:
I'd drop this clear, you don't know if user had anything of value on their screen, didn't ask that to be wiped.